sys/mips/pmap.c

```diff
@@ -552,33 +548,27 @@ void pmap_delete(pmap_t *pmap) {
   pool_free(P_PMAP, pmap);
 }
 
-#define L1_SPACE_SIZE (PAGESIZE * PAGESIZE / sizeof(pte_t))
-
 /*
  * Increase usable kernel virtual address space to at least maxkvaddr.
  * Allocate page table (level 1) if needed.
  */
-vaddr_t pmap_growkernel(vaddr_t maxkvaddr) {
-  assert(maxkvaddr > atomic_load(&pmap_maxkvaddr));
+void pmap_growkernel(vaddr_t maxkvaddr) {
+  assert(mtx_owned(&vm_kernel_end_lock));
+  assert(maxkvaddr > vm_kernel_end);
 
   pmap_t *pmap = pmap_kernel();
   vaddr_t va;
 
-  /*
-   * For 8 pages used by kernel we need 1 page for KASAN.
-   * shadow map is grown by a multiple of L1_SPACE_SIZE.
-   */
-  maxkvaddr = roundup(maxkvaddr, 8 * L1_SPACE_SIZE);
+  maxkvaddr = roundup(maxkvaddr, PAGESIZE);
 
   WITH_MTX_LOCK (&pmap->mtx) {
-    for (va = pmap_maxkvaddr; va <= maxkvaddr; va += L1_SPACE_SIZE) {
+    for (va = vm_kernel_end; va < maxkvaddr; va += PAGESIZE) {
       if (!is_valid_pde(PDE_OF(pmap, va)))
         pmap_add_pde(pmap, va);
     }
-    pmap_maxkvaddr = va;
   }
 
-  /* TODO(pj) add new region into shadow map */
+  vm_kernel_end = va;
```

Comment:
I think `vm_kernel_end = maxkvaddr` is correct. Same for AArch64. It makes no difference as long as the step size in the loop above is the same as the alignment of `maxkvaddr`.

Reply:
As you said it makes no difference. From my point of view step in loop should be always a divider of alignment of `maxkvaddr`. But I have changed that for readability.